### PR TITLE
[dagit] Move Code location and Daemon status polling to context

### PR DESCRIPTION
**Pull request by hellendag**
### Summary & Motivation

We want to be able to show icons on the Deployment tabs, specifically "Code locations" and "Daemons" in open source.

In order to do this, move the status polling behavior to a context value that can be consumed by both the nav icons and the tabs on the Deployment page.

In open source, we'll poll for both `code-locations` and `daemons`. In Cloud, we only want `code-locations` because we never show icons for "Agents".

With flag, plus code location and daemon errors:

Screen Shot 2022-10-04 at 10 30 02 AM

Without flag, with daemon errors:

Screen Shot 2022-10-04 at 10 30 17 AM

### How I Tested These Changes

In open source Dagit, verify that the warning icon and spinner appear correctly in the top nav and on the Deployment page tabs. Verify that this works correctly both with and without the "New workspace page" feature flag.

Repeat in Cloud, verify that Cloud master works correctly with this change in OS.

### CODE REVIEW

**Comment by hellendag** [comment]
Current dependencies on/for this PR:
* master
  * **PR #9883** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9883" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/9883?utm_source=stack-comment).

**Review by salazarm — COMMENTED** [review]
Looks good. Do we want tests for the hooks?

**Comment by hellendag** [comment]
`InstanceWarningIcon`, `WorkspaceStatus`, and `AppTopNav` tests should get us the relevant coverage.

File: js_modules/dagit/packages/core/src/nav/useCodeLocationsStatus.tsx [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+
+const POLL_INTERVAL = 3 * 1000;
+
+export const useCodeLocationsStatus = (skip = false): StatusAndMessage | null => {
+  const {locationEntries, refetch} = React.useContext(WorkspaceContext);
+  const history = useHistory();
+
+  const [showSpinner, setShowSpinner] = React.useState(false);
+
+  const queryData = useQuery<WorkspaceUpdateCheckQuery>(WORKSPACE_UPDATE_CHECK_QUERY, {
```
Comment (bengotow) [inline js_modules/dagit/packages/core/src/nav/useCodeLocationsStatus.tsx:23]: Might be nice to rename this query along with the hook

File: js_modules/dagit/packages/app/src/index.tsx [hunk 2]
```diff
@@ -31,6 +32,7 @@ const config = {
   origin: process.env.REACT_APP_BACKEND_ORIGIN || document.location.origin,
   staticPathRoot: `${pathPrefix}/`,
   telemetryEnabled,
+  statusPolling: new Set<DeploymentStatusType>(['code-locations', 'daemons']),
```
Comment (bengotow) [inline js_modules/dagit/packages/app/src/index.tsx:35]: This looks good to me - I kinda wish cloud could just make the `daemons` status a no-op or quietly do nothing on the server-side so we didn't have to conditionally request the status, but this seems like a good client side approach!

**Review by bengotow — APPROVED** [review]

**Pull request merged**